src/streams/mac_h264.rs

```diff
@@ -0,0 +1,198 @@
+    /// Valid choices here are:
+    ///   InputType::UYVY422, InputType::YUYV422, InputType::NV12,
+    ///   InputType::BGR24, InputType::RGB24
+    pub input_type: InputType,
+
+    pub enc_type: EncoderType,
+    pub ffmpeg_opts: FfmpegOptions,
+}
+
+pub struct MacRawH264Stream {}
+
+impl MacRawH264Stream {
+    /// Spawns a blocking thread that:
+    /// 1. Launches `ffmpeg -f avfoundation … -f rawvideo pipe:1` to capture raw frames.
+    /// 2. Reads exactly `frame_size` bytes per frame (according to `input_type`).
+    /// 3. Calls `H264Encoder::encode_raw(Some(pts), &raw_frame_bytes)`.
+    /// 4. If `Some(EncodedFrame)`, sends `encoded_frame.nal_bytes` via `tx.blocking_send(...)`.
+    ///
+    /// Returns a `ReceiverStream<BytesMut>` that downstream can treat exactly like `V4lH264Stream`.
+    pub fn new(cfg: MacRawH264Config) -> Result<ReceiverStream<BytesMut>> {
+        let (tx, rx) = mpsc::channel::<BytesMut>(10);
+
+        // 1) Determine ffmpeg’s "-pixel_format" string from our InputType:
+        let pix_fmt_str: &'static str = match cfg.input_type {
+            InputType::UYVY422 => "uyvy422",
+            InputType::YUYV422 => "yuyv422",
+            InputType::NV12 => "nv12",
+            InputType::BGR24 => "bgr24",
+            InputType::RGB24 => "rgb24",
+            other => {
+                panic!(
+                    "mac_raw_h264: unsupported InputType::{:?}. \
+                     Only UYVY422, YUYVY422, NV12, BGR24, or RGB24 are allowed here.",
+                    other
+                );
+            }
+        };
+
+        // 2) Compute how many bytes each raw frame is, before encoding.
+        let frame_size: usize = match cfg.input_type {
+            InputType::UYVY422 | InputType::YUYV422 => {
+                // 2 bytes per pixel (4:2:2)
+                (cfg.input_width as usize) * (cfg.input_height as usize) * 2
+            }
+            InputType::NV12 => {
+                // NV12 = Y plane (W*H bytes) + UV plane (W*H/2 bytes) = (3/2) * W*H
+                ((cfg.input_width as usize) * (cfg.input_height as usize) * 3) / 2
+            }
+            InputType::BGR24 | InputType::RGB24 => {
+                // 3 bytes per pixel (packed 24-bit)
+                (cfg.input_width as usize) * (cfg.input_height as usize) * 3
+            }
+            _ => unreachable!("pix_fmt_str match above should have panicked for other variants"),
+        };
+
+        thread::spawn(move || {
+            // 3) Build & spawn the ffmpeg command to capture rawvideo from AVFoundation:
+            // ffmpeg -f avfoundation \
+            //        -pixel_format {pix_fmt_str} \
+            //        -video_size {input_width}x{input_height} \
+            //        -framerate {framerate} \
+            //        -i {device_index} \
+            //        -f rawvideo pipe:1
+            let mut cmd = Command::new("ffmpeg");
+            cmd.arg("-f")
```

Comment:
Lets just make this a StdinH264Stream. Then someone can run this command in a bash pipeline with your ffutility example instead of us doing the ffmpeg exec here. This will then allow us to not just play from the webcam but also play from a file or any other arbitrary video stream on stdin